src/net/relay.rs

```diff
@@ -998,17 +1008,114 @@ impl Relayer {
                 }
             }
 
-            // if we did indeed store this microblock (i.e. we didn't have it), then we can relay it
-            if stored {
+            // If we did store any microblocks (i.e. we didn't have it), then we can relay it
+            if stored_microblocks.len() > 0 {
+                info!(
+                    "Stored {} downloaded microblocks.",
+                    stored_microblocks.len()
+                );
                 let index_block_hash =
                     StacksBlockHeader::make_index_block_hash(consensus_hash, &anchored_block_hash);
-                ret.insert(
+                downloaded_microblock_map.insert(
                     (*consensus_hash).clone(),
-                    (index_block_hash, microblock_stream.clone()),
+                    (index_block_hash, stored_microblocks),
                 );
             }
         }
-        ret
+
+        // Now, process and store microblocks obtained through the microblock tip sync protocol.
+        if let Some(result) = &network_result.synced_microblock_result {
```

Comment:
Do you think this code could be in its own method, like `preprocess_synced_microblocks()`?